state/application.go

```diff
@@ -1645,6 +1601,16 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 			})
 		}
 
+		if cfg.Series != "" {
```

Comment:
Did we need to validate that Series should only be != "" for v2 charms

Reply:
I could add a check for this, but it's not actually an error to set `Series` for a v1 charm (e.g., to "kubernetes"), we just don't do it at the moment. This shouldn't be set by the caller in the v1 case though. So IMO no need for additional validation here.